### ISSUE TITLE
Unsupervised Representation Learning with Deep Convolutional Generative Adversarial Networks

### ISSUE THREAD

**Issue by Yamazakii** [Yamazakii/paper_reading#2]
# どんなもの
## 論文
Unsupervised Representation Learning with Deep Convolutional Generative Adversarial Networks
https://arxiv.org/abs/1511.06434

## 著者・所属機関
Alec Radford & Luke Metz
indico Research

Soumith Chintala
Facebook AI Research

## 投稿日
2016/01/07

# Abstract
- 近年，教師あり学習でのCNNは注目されているが，半教師あり学習でのCNNは比較的注目度が低い．
- deep convolutional generative adversarial networks(DCGANs)を紹介する．
- DCGANsはアーキテクチャーの制約があり，半教師あり学習において有力な候補になることを示した．
- いくつかの画像データセットで実験を行ったところ，DCGANはGeneratorとDiscriminatorの両方からオブジェクトパーツから表現のヒエラルキーを学習できるための，十分な証拠を示した．
- 新たなタスクから学習した特徴を用いて，一般的な画像表現としての適用性を示す．

# 先行研究と比べて何がすごい？
- GANより学習が安定する．
- 学習済みのDiscriminatorを画像分類タスクに用いて，他の教師なし学習アルゴリズムと競合するパフォーマンスであることを示す．
- GANの学習を可視化し，特定のフィルターが特定のオブジェクトを描画できることを経験的に示す．
- Generatorが，生成されたサンプルの多くの意味的な性質を容易に操作できる，興味深いベクトル演算的な性質を持つことを示す．

# 技術や手法の肝はどこ？
- Architecture
![fig1](https://github.com/Yamazakii/image_strage/blob/master/dcgan/fig1.PNG?raw=true "fig1")

- CNNの構造に3つの変化を加えた．

### 1つ目
poolingをストライド2の畳み込みに置き換えたall convolutional net([https://arxiv.org/abs/1412.6806:title])
が，自身の空間的なdownsamplingを学習できるようにしたこと．
この手法をGeneratorに用い，自身の空間アップサンプリングとDiscriminatorを学習できるようにした．(？)

### 2つ目
CNNの最後尾にある全結合層の排除．
Global Average Poolingによって，モデルの安定性が向上した一方で，学習の収束には時間がかかるようになった．
入力としてノイズZを受け取るGANの第1層は，単なる行列の乗算なので全結合しているといえる．
しかし，結果(出力)は4階テンソルにreshapeされ，畳み込みの開始点となる．
Discriminatorに関しては，最後の畳み込み層はflattenされて，単なるsigmoidの出力となる．

### 3つ目
学習を安定させるために，Batch Normalizationを使った．
これによって初期化不良のために学習で生じる問題が解決され，より深いモデルでも勾配が流れるようになった．
Batch Normalizationを全層に適用すると，サンプルの振動(？)や，モデルの不安定性が生じる．
これらの問題は，Generatorの出力層とDiscriminatorの入力層にBatchNormを適用しないことで避けられる．

Generatorの活性化関数はReLUを使用するが，出力層にはtanhを用いる．
Discriminatorでは，Leaky ReLUが良い性能を示した．
これはmaxout activationを使用した，オリジナルのGANの論文とは対照的な結果を示している．

# どうやって有効だと検証した？
## 実験条件
- stochastic gradient decent(SGD)でミニバッチサイズ128で学習．
- 平均0.0，標準偏差0.02のガウス分布(Normal distribution)で初期化．
- LeakyReLUの傾きは0.2．
- 従来のGANはハイパーパラメータのチューニングにmomentumを使用したが，今回はAdamを使用．
- 学習率0.001は大きすぎることが判明したので，0.0002を使用した．
- Adamのハイパーパラメータの[tex:β_1]は，0.9では学習が安定しないので，安定性を上げるために，0.5にした．

![fig3](https://github.com/Yamazakii/image_strage/blob/master/dcgan/fig3.PNG?raw=true "fig3")
いくつかのベッドの下の板のノイズテクスチャが，過小適合を表現してるとのこと．

![fig4](https://github.com/Yamazakii/image_strage/blob/master/dcgan/fig4.PNG?raw=true "fig4")
窓のない部屋に窓が発生したり，テレビが窓に変わったり...

# 議論はある？
- まだ不安定さが残るので，これを解決するためのさらなる研究が必要．
- このフレームワーク(DCGANのこと)を動画(frame prediction)や音(pre-trained features for speech synthesis)に拡張するのは興味深い．
- 学習された潜在空間の性質(properties of the latent space)へのさらなる研究も同様に興味深い．

# 次に読むべき論文は？
CycleGANとか読んでみたい
その他GANの亜種
